test_libs/pyspec/eth2spec/test/helpers/state.py

```diff
@@ -23,3 +26,13 @@ def get_state_root(spec, state, slot) -> bytes:
     """
     assert slot < state.slot <= slot + spec.SLOTS_PER_HISTORICAL_ROOT
     return state.latest_state_roots[slot % spec.SLOTS_PER_HISTORICAL_ROOT]
+
+
+def state_transition_and_sign_block(spec, state, block):
```

Comment:
Verified we are not outputting any blocks in the tests not produced by this function. `appyl_empty_block` is not used for output anywhere. But we may want to remove its return in the future, so we don't accidentally get a block with faulty state root in there.